public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+  - `FormArray`: an array of `FormControls`.
+  - `FormBuilder`: a builder of `FormControls`, `FormGroups`, and `FormArrays`.
+  - `FormControl`: a representation of an HTML form control.
+  - `FormGroup`: a group of `FormControls`, `FormArrays`, and other `FormGroups`.
+  - `Address` and `Hero` are application model classes.
+  - `states`: U.S. states (test data). 
+
+  Now enter the `@Component` decorator that specifies the `HeroDetailComponent` metadata:
+
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'reactive-comp-metadata','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  The `moduleId: module.id` lets you use 
+  [component-relative paths](./cookbook/component-relative-paths.html) in file URLs. 
+  The `selector` tells Angular the name of the element tag that renders this form. 
+  The `templateUrl` is the location of the HTML template that displays the form.
+
+  Next, create an exported `HeroDetailComponent` class with a `FormControl`. 
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v1','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  Inside the `FormGroup`, called `heroForm`, is a `FormControl` called `name`.
+  It will be bound in the template to an HTML `input` box for the hero name. 
+
+  A `FormControl` constructor accepts three, optional arguments: 
+  the initial data value, an array of validators, and an array of async validators.
+  
+  This simple control doesn't have data or validators.
+  In real apps, most form controls have both. 
+
+.l-sub-section
+  :marked
+    Validators won't be covered at all in this guide.
+    Learn about them in the [Form Validation](../cookbook/form-validation.html) cookbook.
+
+.l-main-section
+a#create-template
+:marked
+  ## Create the template
+  Now create the component's template, `app/hero-detail.component.html`, with the following markup.
+
++makeExample('reactive-forms/ts/app/hero-detail-1.component.html', 'basic-form','app/hero-detail.component.html')(format=".")
+
+:marked
+  The `novalidate` attribute in the `<form>` element prevents the browser 
+  from attempting native HTML validations. 
+  The `[formGroup]="heroForm"` tells Angular that the form element
+  is associated with the `heroForm` property of the component class.
+  The `formControlName="name"` associates the `input` element with the `name` control inside the `heroForm`.
+
+.l-sub-section
+  :marked
+    Disregard the `form-group` and `form-control` _CSS_ classes. They belong to the 
+    <a href="http://getbootstrap.com/" target="_blank" title="Bootstrap CSS">Bootstrap CSS library</a>,
+    not Angular.
+    They _style_ the form but in no way impact the logic of the form.
+
+a#import
+:marked
+  ## Import the _ReactiveFormsModule_
+
+  The `HeroDetailComponent` template uses directives from 
+  the `ReactiveFormsModule` in the `@angular/forms` library
```

Comment:
I'm okay with omitting the library part, but we have to be careful to stick with either selectives or class names.  `FormGroup` isn't a directive, it's a class.  So either we want:

- `formGroup` and `formControlName`

OR

- `FormGroupDirective` and `FormControlName`

I think the former is clearer because people don't need to know the formal name of the directive classes.